znnl/training_recording/jax_recording.py

```diff
@@ -618,3 +727,27 @@ def _export_in_memory_data(self) -> dataclass:
         }
 
         return DataSet(**selected_data)
+
+    def _adjust_ntk_shape(self, parsed_data):
```

Comment:
Do you have tests for this? This can probably also be a matrix util but then it should be moved to the utils module and heavily tested.